sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentTests.java

```diff
@@ -115,26 +114,69 @@ public CompletableFuture<TestAsyncResult> runAsync(Supplier<CompletableFuture<Ma
             }
             var mockEngine = (MockEngine) engine;
             var mockMonitor = (MockMonitor) monitor;
-            return this.runner.runAsyncFuture(callback)
+            var helper = new OutputsCaptureHelper(callback);
+            return this.runner.runAsyncFuture(helper::run)
                     .thenApply(ignore -> new TestAsyncResult(
                             ImmutableList.copyOf(mockMonitor.resources),
                             mockEngine.getErrors().stream()
                                     .map(RunException::new)
-                                    .collect(toImmutableList())
-                    ));
+                                    .collect(toImmutableList()),
+                            helper.getOutputs()));
+        }
+
+        public CompletableFuture<TestAsyncResult> tryTestAsync(Consumer<Context> stackFactory) {
+            return this.runAsync(() -> {
+                var context = new TestContext();
+                stackFactory.accept(context);
+                return CompletableFuture.completedFuture(context.getStackOutputs());
+            });
+        }
+
+        public CompletableFuture<ImmutableList<Resource>> testAsync(Consumer<Context> stackFactory) {
```

Comment:
Perhaps `runAsync`?

Reply:
I've reduced the amount of those methods to 1 universal